### ISSUE TITLE
Possible mismatch between Codex and fafbseg.flywire

### ISSUE THREAD

**Issue by adjavon** [navis-org/fafbseg-py#19]
Hi! First of all thanks for making this, it's been extremely useful and the docs are 💯 

I've been having an issue with a subset of my queries by cell type returning errors/being empty. What's surprised me is that for several of these, I can find them through [Codex](https://codex.flywire.ai/app/search?filter_string=AVLP560). I'm using materialization 783 for both as far as I can tell. I've attached a list of cell types for which I'm getting this problem. 
[issues.txt](https://github.com/user-attachments/files/16236430/issues.txt)

Here's what I've been using to query (excuse the giant set of if/try/excepts at the end, I'm trying to make sure I don't miss anything obvious): 
```python
from fafbseg import flywire
from fafbseg.flywire import NeuronCriteria as NC
import logging
from contextlib import redirect_stdout


def get_from_codex(cell_type, regex=False):
    with redirect_stdout(None):
        filter_cell_type = NC(
            cell_type=cell_type,
            regex=regex,
        )
        result_cell_type = flywire.search_annotations(filter_cell_type)
    return result_cell_type


def get_from_hemibrain(cell_type, regex=False):
    with redirect_stdout(None):
        filter_hemibrain = NC(
            hemibrain_type=cell_type,
            regex=regex,
        )
        result_hemibrain = flywire.search_annotations(filter_hemibrain)
    return result_hemibrain


def get_neurons(cell_type):
    result_cell_type = None
    try:
        result_cell_type = get_from_codex(cell_type)
    except:
        logging.info(f"Could not find {cell_type} in Codex cell types")
    if result_cell_type is None:
        try:
            result_cell_type = get_from_hemibrain(cell_type)
        except:
            logging.info(f"Could not find {cell_type} in Hemibrain cell types")
    if result_cell_type is None:
        try:
            result_cell_type = get_from_codex(cell_type, regex=True)
        except:
            logging.info(f"Could not find {cell_type} in Codex cell types using regex")
    if result_cell_type is None:
        try:
            result_cell_type = get_from_hemibrain(cell_type, regex=True)
        except:
            logging.info(
                f"Could not find {cell_type} in Hemibrain cell types using regex"
            )
    return result_cell_type
```

Possibly relevant details: 
```
>>> fafbseg.__version__
'3.0.8'
>>> navis.__version__
'1.6.0'
```

Grateful for any help!

**Comment by schlegelp** [comment]
Hi & thanks for reporting this. I'm currently traveling but I will look into this next early week.

**Comment by schlegelp** [comment]
I haven't checked all the cell types you have flagged but I think the issue is this:

For the structured annotations - i.e. `flywire.search_annotations()` and `flywire.get_hierarchical_annotations()` - `fafbseg` will download and cache data from https://github.com/flyconnectome/flywire_annotations (specifically [this](https://github.com/flyconnectome/flywire_annotations/blob/main/supplemental_files/Supplemental_file1_neuron_annotations.tsv) file). The idea is to use Github for future updates and versioning.

It seems there is a discrepancy between what the function docstring and the tutorials say about the versioning:
- in the functions' docstrings it says it will always use the latest commit on the Github repo.
- in the tutorial it says it will use the latest tagged release

I just had a look at the code and what's happening if you don't specify the annotation version is that it will use  the _latest tagged release_ which is currently "version 2.0.0" from January 2024, i.e. a bit dated.

Originally, the idea was that tagged releases were supposed to be more stable than incremental updates and should therefore be used by default. I'm still a bit on the fence about that

To cut a long story short: by default `flywire.search_annotations()` currently uses old data which explains why your searches came up empty.

I just released a new version (`3.0.9`) which includes a few fixes and makes the search more verbose in terms of annotation version used. 

I will also tag a new release of the annotations ASAP but in the meantime, you can just use `flywire.set_default_annotation_version()` to make sure you are using the up-to-date data:

```python
>>> from fafbseg import flywire 
>>> NC = flywire.NeuronCriteria
>>> flywire.set_default_annotation_version('latest_commit')
>>> flywire.search_annotations(NC(cell_type='AC neuron'))
Using annotation version "latest commit" (b2bceba from 2024-04-23) from https://github.com/flyconnectome/flywire_annotations.
Found 4 neurons matching the given criteria.
Using materialization version 783.

       supervoxel_id             root_id   pos_x  pos_y  pos_z  ...   side  nerve    vfb_id        fbbt_id status
0  80928522772240247  720575940619893072  149775  73234   1985  ...  right     AN  fw015025  FBbt_00007333    NaN
1  78956754960541111  720575940618005531  121674  50932   3057  ...   left     AN  fw047589  FBbt_00007333    NaN
2  78183798151681401  720575940612856854  109897  67399   1788  ...   left     AN  fw047645  FBbt_00007333    NaN
3  80787647845085573  720575940633990812  148029  71184   2057  ...  right     AN  fw047646  FBbt_00007333    NaN
```

Don't forget to update before trying the above though!

Hope this is helpful.

**Comment by adjavon** [comment]
Thanks! I just ran a quick test with and this solves it for the problem cases I had previously. I'll run my full pipeline again during the week and let you know if anything else weird comes up but for now I think it's safe to say the problem's solved.
